### ISSUE TITLE
Taxon Detail screen; Taxonomy section

### ISSUE THREAD

**Issue by jtklein** [inaturalist/iNaturalistReactNative#876]
In the Taxon Detail screen, the Taxonomy section needs to show two subsections, one showing the taxonomy above and including the current taxon, and one showing the taxonomy below (or inside) the current taxon.
Currently, I believe we are only showing the section above and including the current taxon. So, this section only needs to be restyled according to the recent [designs](https://www.figma.com/file/MChpvx4ZrKEVVwsWKt4lkI/iNaturalist-Mobile-UI-Design?type=design&node-id=2320-16604&mode=dev).

If the current taxon is anything other than species, the subsection to show taxa inside the current taxon is not visible but instead a View Children button is displayed. Pressing this button shows this subsection. Collapsed state [here](https://www.figma.com/file/MChpvx4ZrKEVVwsWKt4lkI/iNaturalist-Mobile-UI-Design?type=design&node-id=2703-19005&mode=dev) enlarged state [here](https://www.figma.com/file/MChpvx4ZrKEVVwsWKt4lkI/iNaturalist-Mobile-UI-Design?type=design&node-id=2255-16897&mode=dev).
In other words: For taxonomy, all ranks from Kingdom down to the current taxon’s rank should be displayed, and all children should be hidden behind a “view children taxa” button - except for species - because there could literally be hundreds of children.
Each taxon below the current taxon is listed with an arrow icon that anchors to the first line in multiline text.

Clicking any taxa on this entire list (above and below current taxon) should open that taxon’s taxon detail page.
